browser/src/components/NodeSummary.tsx

```diff
@@ -108,26 +108,53 @@ export function NodeSearch({ graphManager, weightService, nodeName }: SearchProp
       </div>
     )
   }
-  // will partially display each nodes summary
   if (searchResult.length > 1) {
-    {searchResult.map(element => {
-      if(displayNameHelper.displayNameForKey(element.node.key) == nodeName){
-        //TODO: If nodeName is CoffeeMaker && results list contains CoffeeMakerController & CoffeeMaker return CoffeeMaker
-      }
-    })}
-    return (
-      <div>
-        {searchResult.map(element => {
-          var prop : Props = {graphManager, weightService, componentName : element.componentName, nodeName : element.node.key, fullDetails: false}
-          return NodeSummary(prop)
-        })}
-      </div>
-    )
-  }
+    //return fullNode Summary if exactly 1 of the result nodes are equal to the short name of a node 
+    if (shortNameEqualsNodeName(searchResult, nodeName)) {
+      return (
+        <div>
+          {searchResult.map(element => {
+            if(displayNameHelper.displayNameForKey(element.node.key) == nodeName){
+              var prop : Props = {graphManager, weightService, componentName : element.componentName, nodeName : element.node.key, fullDetails: true}
+              return NodeSummary(prop)
+            }
+          })}
+        </div>
+      )
+    } else {
+      // will partially display each nodes summary
+      return (
+        <div>
+          {searchResult.map(element => {
+            var prop : Props = {graphManager, weightService, componentName : element.componentName, nodeName : element.node.key, fullDetails: false}
+            return NodeSummary(prop)
+          })}
+        </div>
+      )
+    }
+  } else {
+  //if searchResult is a length of 1, return the summary of the node
   var componentName = searchResult[0].componentName
   var nodeName: string = searchResult[0].node.key
   var prop: Props = {graphManager, weightService, componentName, nodeName, fullDetails : true}
   return NodeSummary(prop)
+  }
+}
+
+function shortNameEqualsNodeName(searchResult: GraphMatchResult[], nodeName: string) : boolean {
+  const displayNameHelper = new DisplayNameHelper()
+  //counts how many short names in the graph equals to nodeName 
+  var commonCount: number = 0
+  searchResult.map(element => {
```

Comment:
instead of .map(), consider using `.filter()` to get all the results that fit `displayNameHelper.displayNameForKey(element.node.key) == nodeName`, and you can check the result size, instead of defining another variable `commonCount `